### ISSUE TITLE
Unable to import VSCode Extensions & Settings

### ISSUE THREAD

**Issue by twlite** [getcursor/cursor#827]
Hello, I use vscode from the [insiders channel](https://code.visualstudio.com/insiders) and I wanted to try cursor. As suggested, I tried to import my extensions and settings from vscode insiders using this command:

![image](https://github.com/getcursor/cursor/assets/46562212/ea7e95c3-a95a-40ca-8018-2f506c8f281a)

Upon using that command, nothing actually happens. I do not see my settings or extensions in cursor. Is importing from insiders not supported yet?

**Comment by rawic** [comment]
The same thing happens to me. There's another way to import VSC settings by clicking on the settings icon in the top right corner. Then you can click 'Import,' but in my case, the label just changes to 'Done' and nothing actually gets imported.

**Comment by rawic** [comment]
The same thing happens to me. There's another way to import VSC settings by clicking on the settings icon in the top right corner. Then you can click 'Import,' but in my case, the label just changes to 'Done' and nothing actually gets imported.

It's worth noting that this occurs on macOS. On Windows, I haven't noticed any problems (using the Insiders version of VSC as well).

**Comment by rawic** [comment]
The same thing happens to me. There's another way to import VSC settings by clicking on the `settings icon` in the top right corner. Then you can click `Start Import`, but in my case, the label just changes to `Done` and nothing actually gets imported.

In my case the problem occurs on macOS. On Windows, I haven't noticed any problems (using the Insiders version of VSC as well).

**Comment by vavanesssa** [comment]
One easy solution I found is to sync your settings with github, and install a standard stable Visual Studio Code Studio, then sign-in and turn on settings sync. 
After it's done, you can import all the settings in the cursor app.

**Comment by twlite** [comment]
> One easy solution I found is to sync your VSC settings with github, and install a standard stable Visual Studio Code Studio,sign-in, and turn on settings sync. Once it's done, you can import all the settings in the cursor app.

yeah but the issue is with importing settings from vscode insiders.

**Comment by vavanesssa** [comment]
> > One easy solution I found is to sync your VSC settings with github, and install a standard stable Visual Studio Code Studio,sign-in, and turn on settings sync. Once it's done, you can import all the settings in the cursor app.
> 
> yeah but the issue is with importing settings from vscode insiders.
👍🏻 
I know, I'm using insiders too, it's just a temporary workarround

**Comment by alon-parag-taiwa** [comment]
1+ on this, Settings import is not working, and there's no setting sync

**Comment by stabildev** [comment]
please fix

**Comment by dgokcin** [comment]
> Still doesn't work, how I can move??? I have 111 extensions - I am not going to install them from scratch

@AuthorProxy Just export using profiles in vscode and import to cursor.

https://code.visualstudio.com/docs/editor/profiles

**Comment by taylor-lindores-reeves** [comment]
This is an issue with vscode insiders

**Comment by TillMac** [comment]
still cant

**Comment by rapcal** [comment]
Same here - from ~100 extensions, only around 20 are imported on Mac

**Comment by sohaibilyas** [comment]
1. cmd+shift+p or ctrl+shift+p (for windows)
2. export profile
3. save locally
4. open cursor, same command
5. import profile

yes, we still dont have the sync feature :(

**Comment by dghez** [comment]
Same here, i'm using insiders and didn't work

**Comment by xxrjun** [comment]
Same here

**Comment by matheusrocha89** [comment]
I am using mac os and tried the import command; the extensions and configs were not imported. 

I will try one of the approaches I saw here on the page

**Comment by hugoblanc** [comment]
Importing gist exported VSC profile from here solved the issue 
<img width="536" alt="image" src="https://github.com/user-attachments/assets/806b1e25-3701-4c87-83a6-209a0f6e5eac">

**Comment by HereOrCode** [comment]
VS Code export config
Cursor import import config

![](https://github.com/user-attachments/assets/d04f1870-21a0-4bd8-95ce-cc37ded3e396)

**Comment by matheusrocha89** [comment]
> Importing gist exported VSC profile from here solved the issue <img alt="image" width="536" src="https://private-user-images.githubusercontent.com/15015179/380092550-806b1e25-3701-4c87-83a6-209a0f6e5eac.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA4MDUxNDEsIm5iZiI6MTczMDgwNDg0MSwicGF0aCI6Ii8xNTAxNTE3OS8zODAwOTI1NTAtODA2YjFlMjUtMzcwMS00Yzg3LTgzYTYtMjA5YTBmNmU1ZWFjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA1VDExMDcyMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUwYWExY2M3ZWZlYzQxYjA1N2E2YWQ2Yjk3OTAwZDQ5NzYxN2UyZjViNjg1ZWQxMGE4YTU2ZTVjMmUxMWM1MTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Z1nfFkxZS6OwwnDhFoc3umUD7quFQabJPLt-qtr49WY">

Yeah, I had to do the same. The button to import from VSCode in Cursor settings page didn't work for me.

**Comment by tgrushka** [comment]
Geez, year and a half, and unable to select code-insiders as import source. Simple UI option. Basically only folder name is different.

They are probably losing a ton of signups because a huge portion of developers use insiders version. Would probably take a day or two of dev time to implement this simple feature.

**Comment by Nezteb** [comment]
> One easy solution I found is to sync your VSC insiders settings with github, and install a standard stable Visual Studio Code Studio,sign-in, and turn on settings sync. Once it's done, you can import all the settings in the cursor app.

@vavanesssa I wasn't able to get this to work. Each version seems to sync settings separately:

Code - Insiders:
![Image](https://github.com/user-attachments/assets/f3e7fe05-6fdd-4b58-bfc0-b10f33031628)

Code:
![Image](https://github.com/user-attachments/assets/8a9bbef2-f19a-4556-a7c1-0742cc98a90c)

I was able to do the import/export process though.

**Comment by FahadBinHussain** [comment]
1. open vscode
2. ctrl+shift+p
3. >export profile
4. save locally
5. open cursor
6. ctrl+shift+p
7. >Preferences: Open Profiles (UI)
8. Click Down Arrow next to "New Profile" Button
9. Import Profile

works like a charm!

**Comment by DavidBeckham07** [comment]
https://forum.cursor.com/t/migrating-extensions-from-windows-to-cursor-ai-not-working/22493
this article solve it. 
cd ~/.cursor-server 
cp -r ../.vscode-server/extensions ./

**Comment by DavidBeckham07** [comment]
> cd ~/.cursor-server
> cp -r ../.vscode-server/extensions ./

**Comment by ohabash** [comment]
![Image](https://github.com/user-attachments/assets/12af2eee-df77-48d9-9e22-aeaaf1659a15)

Import doesnt work. I am pressing create but it just loads for a second and quits . The profile did not save.

**Comment by AuthorProxy** [comment]
Super Puper Clever AI MOST MODERN editor cannot fix simple bug for two years, which prevent even to give a try to this editor

**Comment by jiaxiangcheng** [comment]
> ![Image](https://github.com/user-attachments/assets/12af2eee-df77-48d9-9e22-aeaaf1659a15)
> 
> Import doesnt work. I am pressing create but it just loads for a second and quits . The profile did not save.
> 
> Seems like the problem is with an extension that fails to install and stops the whole process.
> 
> I could probably fix this with a few more hours but VScode is fine. Will try cursor again later

Same problem on Windows version.

**Comment by liorg-eleos** [comment]
A: This bug exist (for me) also on Vscode, not just in cursor.
B: My (Hacky) solution:

- Activate settings sync
- Paste the profile settings (manually)
- (In cursor): Cmd + Shift + P -> import Vscode Extensions and settings

**Comment by saadfarooq07** [comment]
Salaam brothers

**Comment by dosicker** [comment]
+1 the same issues！！！（my problem was related to importing extensions from VS Code, other shortcuts and snippets nothing.
strongly pls that Cursor will fix it as soon as possible~
---

But the current solution similar to [#issuecomment-2453263329](https://github.com/getcursor/cursor/issues/827#issuecomment-2453263329)

1、VS Code：Export the VS Code Profile and name it "Default" (because when you import the Cursor, the button at the bottom will be replaced with "Replace" instead of "New")；
2、Cursor：Import the Profile config file just exported from VS Code, and then click "Replace"，done~
---

This is not a perfect solution. The best solution is for Cursor to either provide：
a synchronization configuration function (similar to VS Code) or improve the "Import VSCode Extensions and Setting" function.

**Comment by cilasbeltrame** [comment]
Just something to share: I had some old and deprecated extensions on VScode, causing a WARN there. I have uninstalled them on VScode and It went though on cursor with the import native option on cursor:

![Image](https://github.com/user-attachments/assets/8758481b-8a0f-4cbf-966d-ad263336fe1c)

**Comment by matheusrocha89** [comment]
@cilasbeltrame Can you tell what was the extension?

**Comment by matheusrocha89** [comment]
I made mine work. The problem was the `Code Spell Checker` extension, I uninstalled in VSCode and tried the import in cursor and it works.

<img width="815" alt="Image" src="https://github.com/user-attachments/assets/c8054228-c6ad-4b6a-9849-e985d715ee0b" />

**Comment by itsHaseebSaeed** [comment]
It's been over 2 months. I've tried using the cursor multiple times, but this bug keeps me from migrating completely. That's just ridiculous.

**Comment by cilasbeltrame** [comment]
> [@cilasbeltrame](https://github.com/cilasbeltrame) Can you tell what was the extension?

Look for any deprecated and unsupported extensions as of now.

**Comment by cilasbeltrame** [comment]
> I made mine work. The problem was the `Code Spell Checker` extension, I uninstalled in VSCode and tried the import in cursor and it works.
> 
> <img alt="Image" width="815" src="https://private-user-images.githubusercontent.com/3718366/413574813-c8054228-c6ad-4b6a-9849-e985d715ee0b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDAxNDI4ODQsIm5iZiI6MTc0MDE0MjU4NCwicGF0aCI6Ii8zNzE4MzY2LzQxMzU3NDgxMy1jODA1NDIyOC1jNmFkLTRiNmEtOTg0OS1lOTg1ZDcxNWVlMGIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIyMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMjFUMTI1NjI0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTJmZjRiNGRiYzM0NTRmZGU5OTgxMDI5ZDQxZGU0MjFhZDgzOTIxNjJlODZlNGM1NWJiZTliZWMyNDY0ODY0NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.fkjH8SG9GG4IfrR7EiwO9H4nLMQH8L24VWzxq0dCCz4">

Awesome!! This probably should be the main reason.

**Comment by timmyomahony** [comment]
I've given up with Cursor because of the unreliable sync with Vscode. Will use one of the many competitors. It's an absolute essential to be able to import and sync settings between VSCode and Cursor, I can't believe that this is still a problem.

**Comment by Dangles** [comment]
I support the problem. I faced a problem with synchronization of my settings from a file. I am using VS code portable. And the Cursor migration function doesn't work at all for me.

**Comment by ghorbani-mohammad** [comment]
It's really annoying that I can't sync my cursor settings with my vscode settings, to get all my extensions there. Please fix this issue.

**Comment by h3xexe** [comment]
> ![Image](https://github.com/user-attachments/assets/12af2eee-df77-48d9-9e22-aeaaf1659a15)
> 
> Import doesnt work. I am pressing create but it just loads for a second and quits . The profile did not save.
> 
> Seems like the problem is with an extension that fails to install and stops the whole process.
> 
> I could probably fix this with a few more hours but VScode is fine. Will try cursor again later

https://github.com/microsoft/vscode/issues/173327 it is related with this issue.
Fix was disabling the signature validation for me. Add this line to settings.json then import profile:
`"extensions.verifySignature": false,`

**Comment by EggTronic** [comment]
please fix

**Comment by EggTronic** [comment]
> please fix

I fixed this by removing all VsCode extensions with "warning mark" and updating all VsCode extensions to the latest version.

**Comment by matheusrocha89** [comment]
Ok people. It seems that this is not an easy bug to fix because of a lot of scenarios, different environments, and so on. So, I don't think they will fix it soon. 
What I suggest is: 
- Do the profile export in VSCode and import in Cursor (worked for me the first time).
- Find out what extension could be giving warnings and uninstall before trying to use the import button on cursor.

Better try one of these workarounds if you really wanna import your VSCode config to Cursor and start using it. 

That's what I see from here. This issue has been open since August 2023, we are in 2025. Probably is not a priority in the roadmap of the project or because of the different scenarios, it's really hard to track, debug and fix.

**Comment by 92Infinitus92** [comment]
What the actual fuck, for couple of hours I am trying to import my extension from vscode, vscode insiders and so on in cursor and nothing, what the fuck?

**Comment by denniscampos** [comment]
This is the closest thing that worked for me, thanks to this [[article](https://alfredmyers.com/2019/05/31/how-to-copy-visual-code-extensions-to-another-machine/)](https://alfredmyers.com/2019/05/31/how-to-copy-visual-code-extensions-to-another-machine/). I just changed it to `cursor` instead of `code`, so it depends on whether you open your IDE editor with `cursor .` or `code .`

In my case, I have 51 extensions, and about 37 of them were successfully copied.

Step One:
Run this command:
```
code --list-extensions > vscode-extensions.txt
```
**Note:** `code` opens VSCode for me, not Cursor.

Step Two:
Upload this file to a gist and grab the link in `RAW` mode. (Save the link for later — see below.)

Step Three:
In Cursor, add `"extensions.verifySignature": false` under `settings.json`.

Carefully look at the command below, replace the link (from Step Two), and use either `code` or `cursor`, depending on how you are opening your IDE.
```
#!/bin/bash
curl https://gist.githubusercontent.com/[username]/[gistid]/raw/[gistid]/[gist_filename].txt | xargs -L 1 cursor --install-extension
```

**Comment by JohnTheRevelator** [comment]
# Workaround Using Gist Export

## OS
> macOS - Seqouia 15.3.1

## Insiders About

> Version: 1.99.0-insider
Commit: 2c7a0283817355c81e1b106f34d911762950a4d9
Date: 2025-03-14T05:04:14.195Z
Electron: 34.3.2
ElectronBuildId: 11161073
Chromium: 132.0.6834.210
Node.js: 20.18.3
V8: 13.2.152.41-electron.0
OS: Darwin arm64 24.3.0

## Steps I Did
- Open VsCode - Insiders log in with GitHub account and enable Settings Sync
- In VsCode - Standard with the same GitHub account export the desired profile as a Gist
- Click on Open Link when the Gist export finishes
- On the gist webpage:
  - _Optional - log in with GitHub if needed and enable Setting Sync_
  - Switch to Insiders Version
     ![Image](https://github.com/user-attachments/assets/28439707-4ce3-4468-939f-93ef4dc8fc31)
  - Create the imported profile that
- In the Desktop app check if the profile is available and switch to it
- Enjoy

**Comment by Mxctr1** [comment]
I have found a way to fix this. more like a workaround 

Download this extension : https://marketplace.cursorapi.com/items?itemName=BhargavDetroja.export-your-extensions

On the laptop you want to export your extensions export it and send it via email or anyother solutions then on your new pc use the same extensions to import the extensions.

That's it !

**Comment by nicopoore** [comment]
In my case, clicking "Import settings from VSCode" wasn't doing anything on my new computer, but that was because I didn't have VSCode installed in that computer at all - I assumed importing would import from my Github account as it does when you fresh install VSC, but it only imports from your local VSC installation.

So in my case the fix was to install VSC, import from Github, and then importing from VSC to Cursor (from the Cursor settings menu)

Hope this helps someone

**Comment by on3dd** [comment]
What a garbage editor... They managed to break something that worked perfectly in vanilla VSCode...
